appshell/appshell_extensions_gtk.cpp

```diff
@@ -652,69 +657,256 @@ int32 GetPendingFilesToOpen(ExtensionString& files)
 {
 }
 
-GtkWidget* GetMenuBar(CefRefPtr<CefBrowser> browser)
+static GtkWidget* GetMenuBar(CefRefPtr<CefBrowser> browser)
 {
     GtkWidget* window = (GtkWidget*)getMenuParent(browser);
     GtkWidget* widget;
     GList *children, *iter;
+    GtkWidget* menuBar = NULL;
 
     children = gtk_container_get_children(GTK_CONTAINER(window));
     for(iter = children; iter != NULL; iter = g_list_next(iter)) {
         widget = (GtkWidget*)iter->data;
 
-        if (GTK_IS_MENU_BAR(widget))
-            return widget;
+        if (GTK_IS_MENU_BAR(widget)) {
+            menuBar = widget;
+            break;
+        }
     }
 
-    return NULL;
+    g_list_free(children);
+
+    return menuBar;
+}
+
+static int GetPosition(const ExtensionString& positionString, const ExtensionString& relativeId,
+                       GtkWidget* container, NativeMenuModel& model)
+{
+    if (positionString == "" || positionString == "last")
+        return -1;
+    else if (positionString == "first")
+        return 0;
+    else if (!relativeId.empty()) {
+        int relativeTag = model.getTag(relativeId);
+        GtkWidget* relativeMenuItem = (GtkWidget*) model.getOsItem(relativeTag);
+        int position = 0;
+        GList* children = gtk_container_get_children(GTK_CONTAINER(container));
+        GList* iter;
+        for (iter = children; iter != NULL; iter = g_list_next(iter)) {
+            if (iter->data == relativeMenuItem)
+                break;
+            position++;
+        }
+        if (iter == NULL)
+            position = -1;
+        else if (positionString == "before")
+            ;
+        else if (positionString == "after")
+            position++;
+        else if (positionString == "firstInSection") {
+            for (; iter != NULL; iter = g_list_previous(iter)) {
+                if (GTK_IS_SEPARATOR_MENU_ITEM(iter->data))
+                    break;
+                position--;
+            }
+            position++;
+        }
+        else if (positionString == "lastInSection") {
+            for (; iter != NULL; iter = g_list_next(iter)) {
+                if (GTK_IS_SEPARATOR_MENU_ITEM(iter->data))
+                    break;
+                position++;
+            }
+        }
+        else
+            position = -1;
+        g_list_free(children);
+        return position;
+    }
+    else
+        return -1;
 }
 
 int32 AddMenu(CefRefPtr<CefBrowser> browser, ExtensionString title, ExtensionString command,
-              ExtensionString position, ExtensionString relativeId)
+              ExtensionString positionString, ExtensionString relativeId)
```

Comment:
These are actually implementation of interface which is used by all platforms, so changing this would mean changing in all the platforms implementation.
We can take this for later PR.